File: Programmers/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string word) {
    int answer = 0;
    int weights[5] = {781, 156, 31, 6, 1};
    string vowels = "AEIOU";
    
    for (int i = 0; i < word.size(); i++) 
    {
        int idx = vowels.find(word[i]);
        answer += idx * weights[i] + 1;
    }
    
    return answer;
}

File: Programmers/2/84512. 모음 사전/모음 사전.java
class Solution {
    public int solution(String word) {
        int answer = 0;
        char[] vowels = {'A', 'E', 'I', 'O', 'U'};
        int[] weights = {781, 156, 31, 6, 1};

        for (int i = 0; i < word.length(); i++) {
            int idx = "AEIOU".indexOf(word.charAt(i));
            answer += idx * weights[i] + 1;
        }
        
        return answer;
    }
}